packages/website/utils/propsMeta.ts

```diff
@@ -0,0 +1,84 @@
+import path from 'path';
+import {
+  PropComponentDefinition,
+  PropDefinition,
+} from '@contentful/f36-docs-utils';
+const docgen = require('react-docgen-typescript');
+
+// List of props that we do not need to show
+// because they are, basically, inherited props from HTML elements or React
+const PROPS_TO_HIDE = ['ref', 'key', 'style'];
+
+const getPreparedComponent = (
+  component: PropComponentDefinition,
+): PropComponentDefinition => {
+  const resultComponent: PropComponentDefinition = {
+    description: component.description,
+    displayName: component.displayName,
+    props: {},
+  };
+
+  // filter out the props that come from @types/react definitions and/or are HTML elements
+  const filteredProps = Object.values(component.props).filter(
+    (prop) =>
+      !prop.parent?.fileName.includes('@types/react/index.d.ts') &&
+      !PROPS_TO_HIDE.includes(prop.name),
+  );
+
+  filteredProps.map((prop) => {
+    resultComponent.props[prop.name] = {
+      ...prop,
+      // Convert undefined to null for further data serialization
+      parent: (prop.parent ?? null) as PropDefinition['parent'],
+    };
+  });
+
+  return resultComponent;
+};
+
+function getTypescriptMetaInformation(sourcePath) {
+  try {
+    const tsConfigParser = docgen.withCustomConfig(
+      path.resolve('../../tsconfig.json'),
+      {
+        savePropValueAsString: true,
+        shouldExtractLiteralValuesFromEnum: true,
+        shouldExtractValuesFromUnion: true,
+        skipChildrenPropWithoutDoc: false,
+      },
+    );
+
+    const components: PropComponentDefinition[] =
+      tsConfigParser.parse(sourcePath) || [];
+
+    const result: { [key: string]: PropComponentDefinition } = {};
+    components.map((component) => {
+      result[component.displayName] = getPreparedComponent(component);
+    });
```

Comment:
Wdyt about using reduce here?

```suggestion
    const result: { [key: string]: PropComponentDefinition } = components.reduce((acc, component) => {
      return { ...acc, [component.displayName]: getPreparedComponent(component) };
    }, {});
```